pilot/pkg/model/push_context.go

```diff
@@ -763,6 +779,53 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of authentication policies
+func (ps *PushContext) initAuthNPolicies(env *Environment) error {
+	authNPolicies, err := env.List(schemas.AuthenticationPolicy.Type, NamespaceAll)
+	if err != nil {
+		return err
+	}
+
+	sortConfigByCreationTime(authNPolicies)
```

Comment:
I'm not convinced that sorted by creation time is a right approach. It may make it harder to spot the problem. Imo, sorted by name is more stable. Being said that, this is for breaking tie caused by bad configurations in the first place, so it may be fine, but we'll need to remember to document this.

Reply:
This is used in every other pilot CRD.